pinot-plugins/pinot-query-event-listener/src/main/java/org/apache/pinot/plugin/query/event/listener/broker/NoOpBrokerQueryEventListenerFactory.java

```diff
@@ -0,0 +1,47 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.plugin.query.event.listener.broker;
+
+import org.apache.pinot.spi.annotations.queryeventlistener.BrokerEventListenerFactory;
+import org.apache.pinot.spi.annotations.queryeventlistener.BrokerQueryEventListenerFactory;
+import org.apache.pinot.spi.env.PinotConfiguration;
+import org.apache.pinot.spi.queryeventlistener.BrokerQueryEventListener;
+
+@BrokerEventListenerFactory
+public class NoOpBrokerQueryEventListenerFactory implements BrokerQueryEventListenerFactory {
```

Comment:
Why is there a no-op listener factory? Shouldn't there be a single factory which returns either no-op or the real implementation based on some configs?

Reply:
Yeah as discussed offline we don't need a factory here at all and we can just use the class path provided in the configs to identify the BrokerQueryEventListener.